File: gpstaffapp/src/screens/AnnouncementListScreen.js
import React, { useState, useEffect } from 'react';
import { View, Text, FlatList, TouchableOpacity, RefreshControl } from 'react-native';
import { Ionicons } from '@expo/vector-icons';
import API from '../api/client';

export default function AnnouncementListScreen({ navigation }) {
  const [announcements, setAnnouncements] = useState([]);
  const [loading, setLoading] = useState(false);
  const [filterType, setFilterType] = useState('');

  const coachingTypes = ['ALL', 'OFFLINE', 'ONLINE', 'TEST BATCH'];

  useEffect(() => {
    fetchAnnouncements();
  }, [filterType]);

  const fetchAnnouncements = async () => {
    setLoading(true);
    try {
      const endpoint = filterType && filterType !== 'ALL'
        ? `/admin/announcement?coaching_type=${filterType}`
        : '/admin/announcement';
      const res = await API.get(endpoint);
      if (res.data.status) {
        setAnnouncements(res.data.announcements || []);
      }
    } catch (e) {
      console.error(e);
    } finally {
      setLoading(false);
    }
  };

  return (
    <View className="flex-1 bg-slate-50">
      {/* Filter Tabs */}
      <View className="bg-white py-3 px-4 border-b border-gray-100 shadow-sm">
        <FlatList
          horizontal
          showsHorizontalScrollIndicator={false}
          data={coachingTypes}
          keyExtractor={(item) => item}
          renderItem={({ item }) => {
            const active = filterType === item || (item === 'ALL' && !filterType);
            return (
              <TouchableOpacity
                onPress={() => setFilterType(item === 'ALL' ? '' : item)}
                className={`mr-2 px-4 py-1.5 rounded-full border ${
                  active ? 'bg-primary border-primary' : 'bg-gray-50 border-gray-200'
                }`}
              >
                <Text className={`text-xs font-bold ${active ? 'text-white' : 'text-gray-600'}`}>
                  {item}
                </Text>
              </TouchableOpacity>
            );
          }}
        />
      </View>

      {/* Announcements List */}
      <FlatList
        data={announcements}
        keyExtractor={(item) => item.id.toString()}
        contentContainerStyle={{ padding: 16, paddingBottom: 90 }}
        refreshControl={
          <RefreshControl refreshing={loading} onRefresh={fetchAnnouncements} tintColor="#ff7700" />
        }
        ListEmptyComponent={
          !loading && (
            <View className="items-center justify-center py-20">
              <View className="w-16 h-16 rounded-full bg-blue-50 items-center justify-center mb-3">
                <Ionicons name="megaphone-outline" size={32} color="#2b66a2" />
              </View>
              <Text className="text-gray-600 font-bold text-base">No Announcements</Text>
              <Text className="text-gray-400 text-xs mt-1">Tap + to add a new announcement</Text>
            </View>
          )
        }
        renderItem={({ item }) => (
          <TouchableOpacity
            onPress={() => navigation.navigate('EditAnnouncement', { announcementId: item.id })}
            activeOpacity={0.8}
            className="bg-white rounded-2xl p-4 mb-3 border border-gray-100 shadow-sm"
          >
            <View className="flex-row justify-between items-center mb-2.5">
              <View className="flex-row flex-wrap gap-1.5">
                <View className="bg-primary/10 px-2.5 py-0.5 rounded-md">
                  <Text className="text-[11px] font-bold text-primary">{item.usertype}</Text>
                </View>
                <View className="bg-fanta/10 px-2.5 py-0.5 rounded-md">
                  <Text className="text-[11px] font-bold text-fanta">{item.course || 'All'}</Text>
                </View>
                {item.is_schedule == 1 && (
                  <View className="bg-amber-100 px-2 py-0.5 rounded-md">
                    <Text className="text-[10px] font-bold text-amber-700">SCHEDULED</Text>
                  </View>
                )}
              </View>
              <Text className="text-[11px] text-gray-400 font-medium">
                {item.created_at ? new Date(item.created_at).toLocaleDateString() : ''}
              </Text>
            </View>

            <Text className="text-base font-bold text-gray-800 mb-1.5">{item.title}</Text>
            <Text className="text-xs text-gray-500 leading-5" numberOfLines={3}>
              {item.content?.replace(/<[^>]*>?/gm, '')}
            </Text>

            <View className="flex-row items-center justify-between mt-3 pt-3 border-t border-gray-50">
              <View className="flex-row items-center flex-1 mr-2">
                <Ionicons name="people-outline" size={14} color="#9ca3af" />
                <Text className="text-[11px] text-gray-400 font-medium ml-1.5" numberOfLines={1}>
                  {item.coaching_type || 'ALL'} • {item.category || 'All Categories'}
                </Text>
              </View>
              <View className="flex-row items-center">
                <Text className="text-xs font-bold text-primary mr-1">Edit</Text>
                <Ionicons name="chevron-forward" size={14} color="#2b66a2" />
              </View>
            </View>
          </TouchableOpacity>
        )}
      />

      {/* Floating Action Button */}
      <TouchableOpacity
        onPress={() => navigation.navigate('CreateAnnouncement')}
        className="absolute bottom-6 right-6 bg-fanta w-14 h-14 rounded-full items-center justify-center shadow-lg shadow-fanta/50 active:scale-95"
      >
        <Ionicons name="add" size={30} color="#fff" />
      </TouchableOpacity>
    </View>
  );
}

File: gpstaffapp/lib/screens/announcement_list_screen.dart
import 'package:flutter/material.dart';
import 'package:intl/intl.dart';
import '../api/api_client.dart';
import '../models/announcement_model.dart';
import '../theme/app_theme.dart';
import 'create_announcement_screen.dart';
import 'edit_announcement_screen.dart';

class AnnouncementListScreen extends StatefulWidget {
  const AnnouncementListScreen({super.key});

  @override
  State<AnnouncementListScreen> createState() => _AnnouncementListScreenState();
}

class _AnnouncementListScreenState extends State<AnnouncementListScreen> {
  final List<String> _coachingTypes = ['ALL', 'OFFLINE', 'ONLINE', 'TEST BATCH'];
  String _filterType = '';
  List<AnnouncementModel> _announcements = [];
  bool _loading = false;

  @override
  void initState() {
    super.initState();
    _fetchAnnouncements();
  }

  Future<void> _fetchAnnouncements() async {
    setState(() => _loading = true);
    try {
      final dio = ApiClient().dio;
      final endpoint = _filterType.isNotEmpty && _filterType != 'ALL'
          ? '/admin/announcement?coaching_type=$_filterType'
          : '/admin/announcement';

      final res = await dio.get(endpoint);
      if (res.data != null && res.data['status'] == true) {
        final list = res.data['announcements'];
        if (list is List) {
          setState(() {
            _announcements = list.map((e) => AnnouncementModel.fromJson(e)).toList();
          });
        }
      }
    } catch (e) {
      debugPrint('Fetch announcements error: $e');
    } finally {
      if (mounted) setState(() => _loading = false);
    }
  }

  String _formatDate(String? dateStr) {
    if (dateStr == null || dateStr.isEmpty) return '';
    try {
      final dt = DateTime.parse(dateStr);
      return DateFormat('dd/MM/yyyy').format(dt);
    } catch (_) {
      return dateStr;
    }
  }

  @override
  Widget build(BuildContext context) {
    return Scaffold(
      backgroundColor: AppColors.background,
      appBar: AppBar(
        title: const Text('Announcements'),
      ),
      body: Column(
        children: [
          // Filter Tabs Bar
          Container(
            color: Colors.white,
            padding: const EdgeInsets.symmetric(horizontal: 16, vertical: 10),
            child: SingleChildScrollView(
              scrollDirection: Axis.horizontal,
              physics: const BouncingScrollPhysics(),
              child: Row(
                children: _coachingTypes.map((type) {
                  final active = _filterType == type || (type == 'ALL' && _filterType.isEmpty);
                  return Padding(
                    padding: const EdgeInsets.only(right: 8),
                    child: ChoiceChip(
                      label: Text(type),
                      labelStyle: TextStyle(
                        fontSize: 12,
                        fontWeight: FontWeight.bold,
                        color: active ? Colors.white : AppColors.textSecondary,
                      ),
                      selected: active,
                      selectedColor: AppColors.primary,
                      backgroundColor: const Color(0xFFF8FAFC),
                      side: BorderSide(
                        color: active ? AppColors.primary : AppColors.border,
                      ),
                      shape: RoundedRectangleBorder(borderRadius: BorderRadius.circular(20)),
                      onSelected: (_) {
                        setState(() {
                          _filterType = type == 'ALL' ? '' : type;
                        });
                        _fetchAnnouncements();
                      },
                    ),
                  );
                }).toList(),
              ),
            ),
          ),
          const Divider(height: 1, color: AppColors.borderLight),

          // Announcements List
          Expanded(
            child: RefreshIndicator(
              onRefresh: _fetchAnnouncements,
              color: AppColors.fanta,
              child: _loading && _announcements.isEmpty
                  ? const Center(
                      child: CircularProgressIndicator(color: AppColors.fanta),
                    )
                  : _announcements.isEmpty
                      ? ListView(
                          children: const [
                            SizedBox(height: 100),
                            Center(
                              child: Column(
                                mainAxisAlignment: MainAxisAlignment.center,
                                children: [
                                  CircleAvatar(
                                    radius: 36,
                                    backgroundColor: Color(0xFFE0F2FE),
                                    child: Icon(Icons.campaign_outlined, size: 36, color: AppColors.primary),
                                  ),
                                  SizedBox(height: 16),
                                  Text(
                                    'No Announcements',
                                    style: TextStyle(
                                      fontSize: 16,
                                      fontWeight: FontWeight.bold,
                                      color: AppColors.textPrimary,
                                    ),
                                  ),
                                  SizedBox(height: 4),
                                  Text(
                                    'Tap + to add a new announcement',
                                    style: TextStyle(fontSize: 12, color: AppColors.textMuted),
                                  ),
                                ],
                              ),
                            ),
                          ],
                        )
                      : ListView.separated(
                          padding: const EdgeInsets.fromLTRB(16, 16, 16, 90),
                          itemCount: _announcements.length,
                          separatorBuilder: (_, __) => const SizedBox(height: 12),
                          itemBuilder: (context, index) {
                            final item = _announcements[index];

                            return InkWell(
                              onTap: () async {
                                final res = await Navigator.push(
                                  context,
                                  MaterialPageRoute(
                                    builder: (_) => EditAnnouncementScreen(announcementId: item.id),
                                  ),
                                );
                                if (res == true) _fetchAnnouncements();
                              },
                              borderRadius: BorderRadius.circular(20),
                              child: Container(
                                padding: const EdgeInsets.all(16),
                                decoration: BoxDecoration(
                                  color: Colors.white,
                                  borderRadius: BorderRadius.circular(20),
                                  border: Border.all(color: AppColors.borderLight),
                                  boxShadow: [
                                    BoxShadow(
                                      color: Colors.black.withOpacity(0.02),
                                      blurRadius: 6,
                                      offset: const Offset(0, 2),
                                    ),
                                  ],
                                ),
                                child: Column(
                                  crossAxisAlignment: CrossAxisAlignment.start,
                                  children: [
                                    // Top Row: Badges + Date
                                    Row(
                                      mainAxisAlignment: MainAxisAlignment.spaceBetween,
                                      children: [
                                        Wrap(
                                          spacing: 6,
                                          runSpacing: 4,
                                          children: [
                                            // User Type
                                            Container(
                                              padding: const EdgeInsets.symmetric(horizontal: 8, vertical: 3),
                                              decoration: BoxDecoration(
                                                color: AppColors.primary.withOpacity(0.1),
                                                borderRadius: BorderRadius.circular(6),
                                              ),
                                              child: Text(
                                                item.usertype,
                                                style: const TextStyle(
                                                  fontSize: 10,
                                                  fontWeight: FontWeight.bold,
                                                  color: AppColors.primary,
                                                ),
                                              ),
                                            ),
                                            // Course
                                            Container(
                                              padding: const EdgeInsets.symmetric(horizontal: 8, vertical: 3),
                                              decoration: BoxDecoration(
                                                color: AppColors.fanta.withOpacity(0.1),
                                                borderRadius: BorderRadius.circular(6),
                                              ),
                                              child: Text(
                                                item.course ?? 'All',
                                                style: const TextStyle(
                                                  fontSize: 10,
                                                  fontWeight: FontWeight.bold,
                                                  color: AppColors.fanta,
                                                ),
                                              ),
                                            ),
                                            // Scheduled
                                            if (item.isSchedule == 1)
                                              Container(
                                                padding: const EdgeInsets.symmetric(horizontal: 8, vertical: 3),
                                                decoration: BoxDecoration(
                                                  color: Colors.amber.shade100,
                                                  borderRadius: BorderRadius.circular(6),
                                                ),
                                                child: Text(
                                                  'SCHEDULED',
                                                  style: TextStyle(
                                                    fontSize: 10,
                                                    fontWeight: FontWeight.bold,
                                                    color: Colors.amber.shade900,
                                                  ),
                                                ),
                                              ),
                                          ],
                                        ),
                                        Text(
                                          _formatDate(item.createdAt),
                                          style: const TextStyle(fontSize: 11, color: AppColors.textMuted),
                                        ),
                                      ],
                                    ),
                                    const SizedBox(height: 10),

                                    // Title
                                    Text(
                                      item.title,
                                      style: const TextStyle(
                                        fontSize: 15,
                                        fontWeight: FontWeight.bold,
                                        color: AppColors.textPrimary,
                                      ),
                                    ),
                                    const SizedBox(height: 4),

                                    // Clean Content Preview
                                    if (item.cleanContent.isNotEmpty)
                                      Text(
                                        item.cleanContent,
                                        style: const TextStyle(
                                          fontSize: 12,
                                          color: AppColors.textSecondary,
                                          height: 1.4,
                                        ),
                                        maxLines: 3,
                                        overflow: TextOverflow.ellipsis,
                                      ),

                                    const SizedBox(height: 12),
                                    const Divider(height: 1, color: AppColors.borderLight),
                                    const SizedBox(height: 10),

                                    // Bottom Row: Summary & Edit button
                                    Row(
                                      mainAxisAlignment: MainAxisAlignment.spaceBetween,
                                      children: [
                                        Expanded(
                                          child: Row(
                                            children: [
                                              const Icon(Icons.people_outline, size: 14, color: AppColors.textMuted),
                                              const SizedBox(width: 4),
                                              Expanded(
                                                child: Text(
                                                  '${item.coachingType ?? "ALL"} • ${item.category ?? "All Categories"}',
                                                  style: const TextStyle(fontSize: 11, color: AppColors.textMuted),
                                                  maxLines: 1,
                                                  overflow: TextOverflow.ellipsis,
                                                ),
                                              ),
                                            ],
                                          ),
                                        ),
                                        const Row(
                                          children: [
                                            Text(
                                              'Edit',
                                              style: TextStyle(
                                                fontSize: 12,
                                                fontWeight: FontWeight.bold,
                                                color: AppColors.primary,
                                              ),
                                            ),
                                            SizedBox(width: 2),
                                            Icon(Icons.chevron_right, size: 16, color: AppColors.primary),
                                          ],
                                        ),
                                      ],
                                    ),
                                  ],
                                ),
                              ),
                            );
                          },
                        ),
            ),
          ),
        ],
      ),
      floatingActionButton: FloatingActionButton(
        onPressed: () async {
          final res = await Navigator.push(
            context,
            MaterialPageRoute(builder: (_) => const CreateAnnouncementScreen()),
          );
          if (res == true) _fetchAnnouncements();
        },
        backgroundColor: AppColors.fanta,
        elevation: 4,
        shape: const CircleBorder(),
        child: const Icon(Icons.add, color: Colors.white, size: 28),
      ),
    );
  }
}
